graphql/src/subscription/mod.rs

```diff
@@ -57,13 +58,38 @@ where
         deadline: None,
     };
 
-    match *operation {
+    match operation {
         // Execute top-level `subscription { ... }` expressions
-        q::OperationDefinition::Subscription(ref subscription) => {
-            let source_stream = create_source_event_stream(&ctx, subscription)?;
-            let response_stream =
-                map_source_to_response_stream(&ctx, subscription, source_stream, options.timeout)?;
-            Ok(response_stream)
+        q::OperationDefinition::Subscription(q::Subscription { selection_set, .. }) => {
+            let complexity = ctx
+                .root_query_complexity(
+                    sast::get_root_query_type_def(&ctx.schema.document).unwrap(),
+                    selection_set,
+                )
+                .map_err(|e| vec![e])?;
+
+            info!(
+                ctx.logger,
+                "Execute subscription";
+                "query" => ctx.document.format(&Style::default().indent(0)).replace('\n', " "),
+                "complexity" => complexity,
+            );
+
+            match options.max_complexity {
+                Some(max_complexity) if complexity > max_complexity => {
+                    Err(vec![QueryExecutionError::TooComplex(complexity, max_complexity)].into())
```

Comment:
Same here about logging — if we don't log subscriptions that fail early like this yet, I think we should.

Reply:
The error is now being logged on both cases.